plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/totals/shipping/index.tsx

```diff
@@ -1,165 +1,52 @@
 /**
  * External dependencies
  */
-import clsx from 'clsx';
 import { __ } from '@wordpress/i18n';
-import { useState } from '@wordpress/element';
-import { useStoreCart } from '@woocommerce/base-context/hooks';
 import { TotalsItem } from '@woocommerce/blocks-components';
-import type { Currency } from '@woocommerce/types';
-import { ShippingVia } from '@woocommerce/base-components/cart-checkout/totals/shipping/shipping-via';
-import {
-	isAddressComplete,
-	isPackageRateCollectable,
-} from '@woocommerce/base-utils';
-import { CHECKOUT_STORE_KEY } from '@woocommerce/block-data';
-import { useSelect } from '@wordpress/data';
-import { ShippingCalculatorContext } from '@woocommerce/base-components/cart-checkout/shipping-calculator/context';
+import { getCurrencyFromPriceResponse } from '@woocommerce/price-format';
+import { hasShippingRate } from '@woocommerce/base-utils';
+import { useStoreCart } from '@woocommerce/base-context';
 
 /**
  * Internal dependencies
  */
-import { ShippingCalculator } from '../../shipping-calculator';
-import {
-	hasShippingRate,
-	getTotalShippingValue,
-	areShippingMethodsMissing,
-} from './utils';
-import ShippingPlaceholder from './shipping-placeholder';
-import ShippingAddress from './shipping-address';
-import ShippingRateSelector from './shipping-rate-selector';
+import { ShippingVia } from './shipping-via';
+import { ShippingCollaterals } from './shipping-collaterals';
+import { ShippingAddress } from './shipping-address';
+import { renderShippingTotalValue } from './utils';
 import './style.scss';
 
 export interface TotalShippingProps {
-	currency: Currency;
-	values: {
-		total_shipping: string;
-		total_shipping_tax: string;
-	}; // Values in use
-	showCalculator?: boolean; //Whether to display the rate selector below the shipping total.
-	showRateSelector?: boolean; // Whether to show shipping calculator or not.
-	className?: string;
-	isCheckout?: boolean;
+	label?: string;
+	placeholder?: React.ReactNode | null;
```

Comment:
why do we have `| null` for these 2 props? Is there a situation we'd pass null versus this just being undefined which is covered by `?` ?

Reply:
It's the default value if no prop is passed but I guess we can remove that.

Reply:
https://github.com/woocommerce/woocommerce/pull/52313/commits/a2996fdee798a535f3d9290bf16182c59e9a719e